src/main/java/teammates/common/datatransfer/questions/FeedbackRankRecipientsQuestionDetails.java

```diff
@@ -285,6 +314,56 @@ public String getQuestionResultStatisticsCsv(
         return normalisedRankOfResponse;
     }
 
+    /**
+     * Returns list of responses excluding responses given to self.
+     *
+     * @param responses a list of responses
+     * @return list of responses excluding self given responses
+     */
+    private List<FeedbackResponseAttributes> getResponsesExcludingSelf(List<FeedbackResponseAttributes> responses) {
+        List<FeedbackResponseAttributes> responsesExcludingSelf = new ArrayList<FeedbackResponseAttributes>();
+        for (FeedbackResponseAttributes response : responses) {
+            if (!response.giver.equalsIgnoreCase(response.recipient)) {
+                responsesExcludingSelf.add(response);
+            }
+        }
+        return responsesExcludingSelf;
+    }
+
+    /**
+     * Returns the average excluding self response text.
+     * Displays a dash if the user has only self response.
+     *
+     * @param isAvgExclSelfShown if average excluding self to be shown
+     * @param df decimal format
+     * @param recipientRanksExcludingSelf list of recipient ranks excluding self response
+     * @param recipientKey recipient for which average is to be calculated
+     * @return average excluding self text
+     */
+    private String getAverageExcludingSelfText(boolean isAvgExclSelfShown, DecimalFormat df,
+            Map<String, List<Integer>> recipientRanksExcludingSelf, String recipientKey) {
+        List<Integer> ranksExcludingSelf = recipientRanksExcludingSelf.get(recipientKey);
+        if (ranksExcludingSelf == null) {
+            return isAvgExclSelfShown ? "-" : "";
```

Comment:
Is is necessary to check `isAvgExclSelfShown` all the time this method is called? You're already only using the output of this method in places where it already holds true.